File: app/src/main/java/com/handstandsam/shoppingapp/features/home/CategoryViewHolder.java
package com.handstandsam.shoppingapp.features.home;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.handstandsam.shoppingapp.R;
import com.handstandsam.shoppingapp.features.category.CategoryActivity;
import com.handstandsam.shoppingapp.features.category.CategoryPresenter;
import com.handstandsam.shoppingapp.models.Category;

import butterknife.BindView;
import butterknife.ButterKnife;


class CategoryViewHolder extends RecyclerView.ViewHolder {
    @BindView(R.id.text)
    TextView textView;

    @BindView(R.id.image)
    ImageView imageView;

    Category category;

    public CategoryViewHolder(final View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
        itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Context context = itemView.getContext();
                Intent intent = new Intent(context, CategoryActivity.class);
                Bundle extras = new Bundle();
                extras.putSerializable(CategoryPresenter.BUNDLE_PARAM_CATEGORY, category);
                intent.putExtras(extras);
                context.startActivity(intent);
            }
        });
    }

    public void bindData(Category category, int position) {
        this.category = category;

        String imageUrl = category.getImage();
        if (imageUrl != null && !imageUrl.isEmpty()) {
            Glide.with(imageView.getContext()).load(category.getImage()).into(imageView);
        } else {
            itemView.setBackgroundResource(ColorInts.getColor(position));
        }

        textView.setText(this.category.getLabel());
    }
}

File: app/src/main/java/com/handstandsam/shoppingapp/features/home/CategoryViewHolder.kt
package com.handstandsam.shoppingapp.features.home

import android.content.Intent
import android.os.Bundle
import androidx.recyclerview.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.bumptech.glide.Glide
import com.handstandsam.shoppingapp.R
import com.handstandsam.shoppingapp.features.category.CategoryActivity
import com.handstandsam.shoppingapp.features.category.CategoryActivity.Companion.BUNDLE_PARAM_CATEGORY
import com.handstandsam.shoppingapp.models.Category


internal class CategoryViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    private val textView: TextView = itemView.findViewById(R.id.text)

    private val imageView: ImageView = itemView.findViewById(R.id.image)

    lateinit var category: Category

    init {
        itemView.setOnClickListener {
            val context = itemView.context
            val intent = Intent(context, CategoryActivity::class.java)
            val extras = Bundle()
            extras.putSerializable(BUNDLE_PARAM_CATEGORY, category)
            intent.putExtras(extras)
            context.startActivity(intent)
        }
    }

    fun bindData(category: Category, position: Int) {
        this.category = category

        val imageUrl = category.image
        if (!imageUrl.isEmpty()) {
            Glide.with(imageView.context).load(category.image).into(imageView)
        } else {
            itemView.setBackgroundResource(ColorInts.getColor(position))
        }

        textView.text = this.category.label
    }
}